coriolis/tests/cron/test_cron.py

```diff
@@ -0,0 +1,501 @@
+    def test_start_on_error(
+        self,
+        mock_send_status,
+        mock_exc_info,
+        mock_LOG
+    ):
+        mock_exc_info.return_value = "mock_exc_info"
+        self.cron._func.side_effect = Exception('err_msg1')
+        self.cron._on_error.side_effect = Exception('err_msg2')
+
+        self.cron.start()
+
+        self.cron._func.assert_called_once_with(
+            *self.cron._args, **self.cron._kw)
+        self.cron._on_error.assert_called_once_with("mock_exc_info")
+        mock_send_status.assert_called_once_with(
+            None,
+            {"result": None,
+             "description": self.cron._description,
+             "name": self.cron.name,
+             "error_info": "mock_exc_info"}
+        )
+        mock_LOG.assert_has_calls([
+            mock.call.exception(self.cron._func.side_effect),
+            mock.call.exception(self.cron._on_error.side_effect)
+        ])
+
+
+@ddt.ddt
+class CronTestCase(test_base.CoriolisBaseTestCase):
+    """Test suite for the Coriolis Cron."""
+
+    @mock.patch.object(schemas, 'validate_value')
+    def setUp(self, *_):
+        super(CronTestCase, self).setUp()
+        self.cron = cron.Cron()
+
+    @mock.patch.object(schemas, 'validate_value')
+    def test_register(self, *_):
+        job = cron.CronJob(
+            mock.sentinel.name,
+            mock.sentinel.description,
+            mock.sentinel.schedule,
+            False,
+            datetime.datetime.fromisoformat('2099-01-01'),
+            mock.Mock(),
+            mock.Mock(),
+            mock.Mock(),
+            *[],
+            **{}
+        )
+        self.cron.register(job)
+        self.assertEqual(
+            self.cron._jobs[mock.sentinel.name],
+            job
+        )
+
+    def test_register_no_job(self, *_):
+        self.assertRaises(
+            ValueError,
+            self.cron.register,
+            None
+        )
+
+    def test_unregister(self):
+        self.cron._jobs = {
+            'job_1': 'mock_job',
+            'job_2': 'mock_job'
+        }
+        self.cron.unregister('job_1')
+        self.assertEqual(
+            {'job_2': 'mock_job'},
+            self.cron._jobs
+        )
+
+    def test_unregister_jobs_with_prefix(self):
+        self.cron._jobs = {
+            'pre1_job_1': 'mock_job',
+            'pre1_job_2': 'mock_job',
+            'pre2_job_3': 'mock_job'
+        }
+        self.cron.unregister_jobs_with_prefix('pre1')
+        self.assertEqual(
+            {'pre2_job_3': 'mock_job'},
+            self.cron._jobs
+        )
+
+    @mock.patch.object(eventlet, 'spawn')
+    def test_check_jobs(self, mock_spawn):
+        mock_job = mock.Mock()
+        mock_job2 = mock.Mock()
+        mock_job2.should_run.return_value = False
+        self.cron._jobs = {
+            'job1': mock_job,
+            'job2': mock_job2,
+        }
+
+        self.cron._check_jobs()
+
+        mock_job.should_run.assert_called_once()
```

Comment:
Since both jobs actually call for `should_run`, this is irrelevant. Please remove